File: server/routes/inquiry.js
import express from 'express';
import { readDb, writeDb } from '../database/dbHelper.js';
import authMiddleware from '../middleware/auth.js';

const router = express.Router();

// Submit inquiry Route
router.post('/', async (req, res) => {
  const { name, email, company, phone, message } = req.body;

  if (!name || !email) {
    return res.status(400).json({ message: 'Name and email are required fields.' });
  }

  try {
    const db = await readDb();

    const newInquiry = {
      id: Date.now().toString(),
      name,
      email,
      company: company || '',
      phone: phone || '',
      message: message || '',
      createdAt: new Date().toISOString()
    };

    db.inquiries.push(newInquiry);
    await writeDb(db);

    res.status(201).json({ message: 'Inquiry submitted successfully.', inquiry: newInquiry });
  } catch (error) {
    console.error('Inquiry submission error:', error);
    res.status(500).json({ message: 'Server error during inquiry submission.' });
  }
});

// Fetch all inquiries Route (Protected for authenticated users)
router.get('/', authMiddleware, async (req, res) => {
  try {
    const db = await readDb();
    res.json(db.inquiries);
  } catch (error) {
    console.error('Fetch inquiries error:', error);
    res.status(500).json({ message: 'Server error fetching inquiries.' });
  }
});

export default router;

File: server/routers/inquiry.py
import time
from fastapi import APIRouter, HTTPException, Depends, status
from pydantic import BaseModel, Field
from database import read_db, write_db
from auth import get_current_user

router = APIRouter(prefix="/api/inquiry", tags=["inquiry"])

class InquirySchema(BaseModel):
    name: str = Field(..., min_length=1)
    email: str = Field(..., min_length=1)
    company: str = ""
    phone: str = ""
    message: str = ""

@router.post("")
def submit_inquiry(inquiry_data: InquirySchema):
    name = inquiry_data.name.strip()
    email = inquiry_data.email.strip()
    company = inquiry_data.company.strip()
    phone = inquiry_data.phone.strip()
    message = inquiry_data.message.strip()

    if not name or not email:
        raise HTTPException(
            status_code=status.HTTP_400_BAD_REQUEST,
            detail="Name and email are required fields."
        )

    db = read_db()
    
    new_inquiry = {
        "id": str(int(time.time() * 1000)),
        "name": name,
        "email": email,
        "company": company,
        "phone": phone,
        "message": message,
        "createdAt": time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime())
    }

    if "inquiries" not in db:
        db["inquiries"] = []
    
    db["inquiries"].append(new_inquiry)
    write_db(db)

    return {"message": "Inquiry submitted successfully.", "inquiry": new_inquiry}

@router.get("")
def get_inquiries(current_user: dict = Depends(get_current_user)):
    db = read_db()
    return db.get("inquiries", [])
